src/lib/planviz/ast_visualizer.cpp

```diff
@@ -44,8 +59,13 @@ void ASTVisualizer::visualize(const std::vector<std::shared_ptr<AbstractASTNode>
 }
 
 void ASTVisualizer::_visualize_subtree(const std::shared_ptr<AbstractASTNode>& node, std::ofstream& file) {
+  if (_visited_subtrees.find(node) != _visited_subtrees.end()) {
+    return;
+  }
+  _visited_subtrees.emplace(node);
```

Comment:
If you used the return value of emplace, you could get rid of the check above.

Reply:
yep